#!/usr/bin/env python

PACKAGE = "jinho"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


# ROI cfg parameters
gen.add("x_min", double_t, 0, "x_min_value", 0.2, -200, 10)
gen.add("x_max", double_t, 0, "x_max_value", 12.0, -10, 200)
gen.add("y_min", double_t, 0, "y_min_value", -8.0, -200, 10)
gen.add("y_max", double_t, 0, "y_max_value", 8.0, -10, 200)
gen.add("z_min", double_t, 0, "z_min_value", -0.5, -2, 0)
gen.add("z_max", double_t, 0, "z_max_value", -0.1, -2, 100)

# DBScan cfg parameters
gen.add("minPoints", int_t, 0, "minPoints_value", 1, 1, 100)
gen.add("epsilon", double_t, 0, "epsilon_value", 0.7, 0, 2)
gen.add("minClusterSize", double_t, 0, "minClusterSize_value", 1, 0, 100)
gen.add("maxClusterSize", double_t, 0, "maxClusterSize_value", 8200, 0, 10000)

# Object radius weight cfg parameters
gen.add("objectRadiusWeight", double_t, 0, "objectRadiusWeight_value", 0.58, 0, 3)
gen.add("pivotX", double_t, 0, "pivotX_value", 0.0, -10, 10)
gen.add("pivotY", double_t, 0, "pivotY_value", 0.0, -10, 10)

exit(gen.generate(PACKAGE, "jinho_node", "jinho"))
